tests/check/check_h26xsigned_auth.c

```diff
@@ -2158,12 +2152,18 @@ START_TEST(no_public_key_in_sei_and_bad_public_key_on_validation_side)
   ck_assert_int_eq(sv_rc, SV_OK);
   ck_assert(!auth_report);
   sv_rc = signed_video_add_nalu_and_authenticate(sv_vms, i_nalu->data, i_nalu->data_size, &auth_report);
-  ck_assert_int_eq(sv_rc, SV_EXTERNAL_ERROR);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // TODO:
+  ck_assert_str_eq(auth_report->latest_validation.validation_str, "UP");
 
   // Free nalu_list_item and session.
+  free(bad_private_key);
   nalu_list_free_item(sei);
   nalu_list_free_item(i_nalu);
   signed_video_free(sv_vms);
+  signed_video_free(sv_camera);
+  free(private_key);
 
   // Ownership of |nalu_to_prepend.nalu_data| has been transferred. Do not free memory
 
```

Comment:
As mentioned in the first review. To reduce the code footprint and minimize bugs you can add a helper function for the camera side and one for the validation side and/or combine all tests to one test where you loop through the different scenarios.